tests/_openmp_test_helper/nested_prange_blas.pyx

```diff
@@ -25,18 +41,25 @@ def check_nested_prange_blas(double[:, ::1] A, double[:, ::1] B, int nthreads):
 
         int i
         int prange_num_threads
+        int *prange_num_threads_ptr = &prange_num_threads
 
-    threadpool_infos = None
+    threadpool_infos = [None]
 
-    for i in prange(n_chunks, num_threads=nthreads, nogil=True):
-        dgemm(trans, no_trans, &n, &chunk_size, &k,
-              &alpha, &B[0, 0], &k, &A[i * chunk_size, 0], &k,
-              &beta, &C[i * chunk_size, 0], &n)
+    with nogil, parallel(num_threads=nthreads):
+        if openmp.omp_get_thread_num() == 0:
+            with gil:
```

Comment:
I figured out what was the cause of the hanging. Note exactly the cause but at least which line of the code was faulty. It happens if we try to call `threadpool_info` in another thread than the main thread (I was doing that in a previous commit).

I don't know why but it's probably related to the bad state of OpenMP threadpool in non main thread when there are multiple openmp loaded.

Anyway, it's unrelated to this PR and should be investigated separately.

Reply:
I agree. I see that you opened #27 to track this issue.